transformer/Transform.go

```diff
@@ -0,0 +1,228 @@
+package transformer
+
+import (
+	"bufio"
+	"strings"
+	"fmt"
+	"os"
+	"io/ioutil"
+	"gopkg.in/yaml.v2"
+	"log"
+
+	"github.com/cloud66/starter/common"
+)
+
+//main transformation format function
+func Transformer(filename string, formatTarget string, directlyTransformed bool) (bool, error) {
+
+	var err error
+	_, err = os.Stat(formatTarget)
+
+	// create file if not exists
+	if os.IsNotExist(err) {
+		var file, err = os.Create(formatTarget)
+		checkError(err)
+		defer file.Close()
+	} else {
+		common.PrintError("File %s already exists. Will be overwritten.\n", formatTarget)
+	}
+
+
+	yamlFile, err := ioutil.ReadFile(filename)
+
+	dockerCompose := Docker_compose{
+		Services: make(map[string]docker_Service),
+		Version:  "",
+	}
+
+	serviceYaml := Serviceyml{
+		Services: make(map[string]common.Service),
+		Dbs:      make([]string, 0),
+	}
+
+	if err := yaml.Unmarshal([]byte(yamlFile), &dockerCompose); err != nil {
+		fmt.Println(err.Error())
+	}
+
+	if len(dockerCompose.Services) == 0 {
+		d := make(map[string]docker_Service)
+		err = yaml.Unmarshal([]byte(yamlFile), &d)
+		checkError(err)
+
+		serviceYaml.Services, serviceYaml.Dbs = copyToServiceYML(d, directlyTransformed)
+
+	} else {
+
+		serviceYaml.Services, serviceYaml.Dbs = copyToServiceYML(dockerCompose.Services, directlyTransformed)
+	}
+	if len(serviceYaml.Dbs) != 0 {
+		if serviceYaml.Dbs[len(serviceYaml.Dbs)-1] == "" {
+			serviceYaml.Dbs = serviceYaml.Dbs[:len(serviceYaml.Dbs)-1]
+		}
+	}
+
+	file, err := yaml.Marshal(serviceYaml)
+
+	err = ioutil.WriteFile("service.yml", file, 0644)
+	if err != nil {
+		log.Fatalf("ioutil.WriteFile: %v", err)
+	}
+
+	service_yml, er := os.OpenFile(formatTarget, os.O_RDWR, 0644)
+	checkError(er)
+
+	// write some text to file
+	_, err = service_yml.WriteString(string(file))
+	checkError(err)
+
+	// save changes
+	err = service_yml.Sync()
+	checkError(err)
+
+	service_yml.Close()
+
+	//format long syntax ports
+	service_yml, _ = os.Open(formatTarget)
+	defer service_yml.Close()
+
+	var lines []string
+	scanner := bufio.NewScanner(service_yml)
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+
+	//final format for ENV_VARS, CPU and PORTS
+	text := finalFormat(lines)
```

Comment:
Don't use this - instead change CPUs in `DockerCompose` to float and in `ServiceYml` to int and in `copyToServiceYML` perform the transformation logic instead of writing the file, then reading it, and then writing again.